packages/amplify-cli/src/commands/push.ts

```diff
@@ -41,6 +41,9 @@ async function syncCurrentCloudBackend(context: $TSContext) {
 export const run = async (context: $TSContext) => {
   try {
     context.amplify.constructExeInfo(context);
+    if (context.exeInfo.localEnvInfo.noUpdateBackend) {
+      throw new ConfigurationError('This local environment was configured to not publish backend updates.');
```

Comment:
How about "The local environment configuration does not allow backend updates."?

cc @renebrandel